ARCHITECTURE.md

```diff
@@ -112,37 +120,113 @@ during lookup.
 
 ### `internal/config`
 
-TODO: backwards compat, loading, overrides, ...
+The `config` package implements support for a simple YAML-based configuration
+format for `gopass`. Most of the code in this package is for backwards
+compatibility. Whenever we introduce or remove a config option we need to
+introduce a new fallback version that is automatically attempted when loading
+a config file. To resolve ambiguities when parsing different config versions
+we use a "catch-all" field to catch any unused keys and check that this is
+empty after parsing - otherwise we need to try a different config version.
+
+NOTE: We did support nested configurations for sub-stores but removed this
+because the maintenance cost did not justify the benefits of this feature.
 
 ### `internal/cui`
 
-TODO: history, issues, outlook
+The name `cui` is an abbreviation for `console-user-interface` and contains
+several helper functions to interact with humans over a text based interface.
 
-### `internal/editor`
+Most of these ask the user to select some item from a selection or provide
+some input.
 
-TODO: security
+NOTE: We used to support rich terminal UIs with arrow navigation and such.
+However all existing libraries that were available without CGO were either
+abandoned or buggy on some platforms and we didn't have any capacity to fix
+them. So we had to remove support for this feature.
 
 ### `internal/queue`
 
-TODO: motivation, usage
+The `queue` package implements a FIFO queue that executed strictly sequential
+in the background. This allows for certain operations, like a git push, to be
+taken out of the critical path wrt. user interactions. The queue will be fully
+processed before the process exits.
+
+### `internal/store/root`
 
-### `internal/store`
+The `root store` package implements an internal password store API that (only)
+supports mounting `leaf` stores. It will forward (almost) all operations to
+it's `leaf` stores (cross stores moves being a notable exception) and do the
```

Comment:
"it's" -> "its" `leaf` stores.   
"cross stores moves " -> "moves across stores"